src/calendar.cpp

```diff
@@ -260,7 +260,40 @@ float calendar::sunlight() const
     }
 }
 
+std::string calendar::print_duration( int turns )
```

Comment:
This function looks very similar to `calendar::textify_period`, maybe those should be combined?

There is also similar code to print the time for constructions (`construction::get_time_string`) and for crafting (`recipe::print_time`) that could be replaced by this function.